pkg/discover/discover.go

```diff
@@ -113,6 +113,11 @@ func discover(ctx context.Context, path string) error {
 								Enabled: true,
 							}
 						} else {
+							// copy the Konflux attribute as this is a pointer, and
+							// it would otherwise update the existing branch config
+							v := *other.Konflux
+							other.Konflux = &v
```

Comment:
Perhaps we should do the same with `other.Prowgen` here? and also on line https://github.com/openshift-knative/hack/pull/384/files#diff-493c512c6574cbf0c3e3e1a06288a2f10b8f8124cb27b95a29cf8ef4e48b6f0fR139 ?

Reply:
or maybe worth looking at some deep copy libraries?

Reply:
:+1: 
updated